src/vcpkg/commands.build.cpp

```diff
@@ -617,10 +616,7 @@ namespace vcpkg
                              action.spec.triplet(),
                              action.public_abi(),
                              fspecs_to_pspecs(find_itr->second));
-        if (const auto p_ver = build_info.version.get())
-        {
-            bpgh.version = *p_ver;
```

Comment:
This records the fetched source version when users use `--head`: https://github.com/microsoft/vcpkg/blob/0e2a0e0ad1a370f85a637452be52e2ddf2dcac0b/scripts/build_info.cmake#L44-L46

Arguably this should probably be a combination (`+=`) and not a replacement